### DISCUSSION TITLE
AMD AMF - Remove B-frame related options from the settings because...

### DISCUSSION THREAD

**Discussion by lextra2** [Xaymar/obs-stream-effects#839] [Ideas & Feedback]
### Explain the Feature/Enhancement

Because no relevant AMD GPU supports them.
The last GPU series that supported this was I believe the 300 series, which came out in 2015. That was 7 years ago.

This may sound bold but, I bet that everyone that ever tweaked the b-frame related options never even used a GPU that supported those features. Which means having these settings displayed to the user is pure placebo.

B-frame related settings I'm talking about:
![Capture](https://user-images.githubusercontent.com/20690384/161555873-7922593c-b721-4b9e-abdf-cf05ff7a63a6.PNG)

### What makes this feature/enhancement useful for StreamFX and its users?

Settings which do literally nothing for pretty much every user is misleading and pure placebo.

### Checklist

- [ ] I am offering a monetary bounty/reward for the implementation of this Feature/Enhancement.
- [ ] There is no other way to currently do this in StreamFX or OBS Studio.

**Comment by Xaymar** [comment]
`-bf` is an option in FFmpeg and will be kept as a visual option. Whether your GPU supports it or not is not something StreamFX is concerned with.

**Comment by lextra2** [comment]
For anyone who reads this, you can check a video file for B frames with
`ffprobe.exe -show_frames -select_streams v:0 -show_entries frame=pict_type input.mkv`

@Xaymar 
There are plenty of more options for amf which you do not expose in OBS. What is your point here?
How can settings which literally do nothing but confuse the user be none of your concern?

**Comment by Xaymar** [comment]
> There are plenty of more options for amf which you do not expose in OBS. What is your point here?

Both the AMD AMF (via FFmpeg) and the NVIDIA NVENC (via FFmpeg) were integrated with the exact same rules and logic, and are both based on FFmpeg 4.2 - which is what OBS Studio shipped until 27.2. NVIDIA NVENC (via FFmpeg) recently got a refresh to FFmpeg 4.4+ due to it no longer working as expected without Custom Settings being applied, and now works with both FFmpeg 4.2 and 4.4+.

If you take a look at the code, you'll find that for both encoders, the majority of settings present in FFmpeg 4.2 are exposed. In fact you'll find that the AMF (via FFmpeg) integration includes a higher percentage of settings compared to NVIDIA NVENC (via FFmpeg), as the latter is continuously maintained by NVIDIA, unlike the AMD AMF integration.

> How can settings which literally do nothing but confuse the user be none of your concern?

Again, just because your particular hardware lacks support for it does not guarantee that all hardware lacks support for it. It doesn't matter how old the hardware may be, fact is that the FFmpeg code is reading the setting and tries to apply it, therefore it stays. Whether a user is confused or not is not my problem, that's the user's problem entirely - they could have educated themselves about the hardware and software they use, but chose not to.

**Comment by Xaymar** [comment]
#### Rejected: Does not enhance or improve StreamFX
The suggestion neither improves or enhances StreamFX in any meaningful way.

**Comment by lextra2** [comment]
Having settings that do nothing is meaningful, gotcha!

**Comment by Xaymar** [comment]
You're always free to make your own fork and implement all the "important" changes you deem necessary.

**Comment by Xaymar** [comment]
This request is even less useful now that modern AMF devices support B-Frames again.
